src/classes/HH_HouseholdNaming.cls

```diff
@@ -432,13 +432,13 @@ public without sharing class HH_HouseholdNaming {
                             hhlist.add(h.id);
                     }
                 }
-                if (!hhlist.isEmpty()){
+                if (hhlist.size() == 1 || System.isFuture() || System.isBatch() || System.isQueueable()) {
                     HH_HouseholdNaming hn = new HH_HouseholdNaming();
                     hn.UpdateNames(hhlist);
+                } else if (!hhlist.isEmpty()) {
+                    HH_HouseholdNaming.FutureUpdateNames(hhlist);
```

Comment:
It looks like there is an existing static method in `ACCT_IndividualAccounts_TDTM` called `renameHHAccounts` that does this also: https://github.com/SalesforceFoundation/Cumulus/blob/c8702e2ae8c4a44304c5f692f95e168a7b5e3486/src/classes/ACCT_IndividualAccounts_TDTM.cls#L928-L939 

What do you think about moving that method to this class and calling it here to avoid duplicating the logic?  I think this class might be a better home for that `renameHHAccounts` method but curious to hear your thoughts!